packages/types/mongoose-store/index.ts

```diff
@@ -0,0 +1,64 @@
+import * as Promise from 'bluebird';
+import * as mongoose from 'mongoose';
+
+export default class MongooseStore<T>{
+  constructor(protected conn: mongoose.Connection) {
+
+  }
+}
+
+type ID = string | number;
+
+export interface Store<T> {
+  query(): ExecutableChainQuery<T>;
+  or(): ChainQuery<T>;
+  find(query: QueryParams<T>): Promise<T[]>;
+  create(data: T): Promise<T>;
+  update(id: ID, data: T): Promise<T>;
+  delete(id: ID): Promise<T>;
+}
+
+interface Predicate<T> {
```

Comment:
-1 for that. I think that this can introduce more problems than positive values.
MongoDB definitions query seems to be object: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mongodb/index.d.ts#L471-L473 

I think we can leave that in case we will need  to be fh.db compliant, but if not I think we should avoid building our own query language and use mongodb native query.